src/openrct2/CmdlineSprite.cpp

```diff
@@ -224,12 +224,50 @@ static bool SpriteImageExport(const rct_g1_element& spriteElement, const char* o
     }
 }
 
+static std::optional<GamePalette> PaletteImageImport(const char* path)
+{
+    try
+    {
+        auto format = IMAGE_FORMAT::PNG_32;
+        auto image = Imaging::ReadFromFile(path, format);
+
+        auto width = image.Width;
+        auto height = image.Height;
+        if (width * height != 256)
+        {
+            throw "Custom palette image is not 256 pixels";
+        }
+        GamePalette newPalette;
+        auto pixels = image.Pixels.data();
+        auto src = pixels;
+        auto palIndex = 0;
+        for (uint32_t y = 0; y < height; y++)
```

Comment:
This looks wrong. You should be reading the png directly into the palette.

Reply:
This code is primarily copied from the image import code, so I don't know what you are looking for in it.